lib/tx_build/spec.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Stellar.TxBuild.Spec do
+  @moduledoc """
+  Defines the contracts to build a Stellar transaction.
+  """
+  alias Stellar.TxBuild.{Account, Memo, Operation, Signature, TimeBounds}
+
+  @type opts :: Keyword.t()
+  @type tx :: struct()
+  @type account :: Account.t()
+  @type memo :: Memo.t()
+  @type time_bounds :: TimeBounds.t()
+  @type signatures :: Signature.t() | list(Signature.t())
+  @type operations :: Operation.t() | list(Operation.t())
```

Comment:
These two types seem odd by defining a list and a single element. Why not just a list with a single element?.

Reply:
@luishurtado the `add_operation/1` function is able to receive an operation or a list of operations. 
Same case for signatures!

```elixir
tx =
  account
  |> TxBuild.new()
  |> TxBuild.add_operation(op1)
  |> TxBuild.add_operation([op2, op3])
```

Best!

Reply:
@juanhurtado10, it still seems odd. `add_operation` is singular, so passing a single operation is ok. If we also support several operations, we should provide a `add_operations` (plural) function. In any case, only with the later (`add_operations` function), the code works because we can pass a list with a single operation for the one operation case.

```elixir
tx =
  account
  |> TxBuild.new()
  |> TxBuild.add_operations([op1])
  |> TxBuild.add_operations([op2, op3])
```

Reply:
Changes applied!